cpu/kinetis_common/mcg.c

```diff
@@ -0,0 +1,468 @@
+#define KINETIS_MCG_FLL_FACTOR_1280       (MCG_C4_DRST_DRS(1))
+#define KINETIS_MCG_FLL_FACTOR_1464       (MCG_C4_DRST_DRS(1) | MCG_C4_DMX32_MASK)
+#define KINETIS_MCG_FLL_FACTOR_1920       (MCG_C4_DRST_DRS(2))
+#define KINETIS_MCG_FLL_FACTOR_2197       (MCG_C4_DRST_DRS(2) | MCG_C4_DMX32_MASK)
+#define KINETIS_MCG_FLL_FACTOR_2560       (MCG_C4_DRST_DRS(3))
+#define KINETIS_MCG_FLL_FACTOR_2929       (MCG_C4_DRST_DRS(3) | MCG_C4_DMX32_MASK)
+
+#ifndef KINETIS_MCG_DCO_RANGE
+#define KINETIS_MCG_DCO_RANGE             (48000000U)
+#endif
+
+/* Default DCO_RANGE should be defined in periph_conf.h */
+#if (KINETIS_MCG_DCO_RANGE == 24000000U)
+#define KINETIS_MCG_FLL_FACTOR_FEI        KINETIS_MCG_FLL_FACTOR_640
+#define KINETIS_MCG_FLL_FACTOR_FEE        KINETIS_MCG_FLL_FACTOR_732
+#elif (KINETIS_MCG_DCO_RANGE == 48000000U)
+#define KINETIS_MCG_FLL_FACTOR_FEI        KINETIS_MCG_FLL_FACTOR_1280
+#define KINETIS_MCG_FLL_FACTOR_FEE        KINETIS_MCG_FLL_FACTOR_1464
+#elif (KINETIS_MCG_DCO_RANGE == 72000000U)
+#define KINETIS_MCG_FLL_FACTOR_FEI        KINETIS_MCG_FLL_FACTOR_1920
+#define KINETIS_MCG_FLL_FACTOR_FEE        KINETIS_MCG_FLL_FACTOR_2197
+#elif (KINETIS_MCG_DCO_RANGE == 96000000U)
+#define KINETIS_MCG_FLL_FACTOR_FEI        KINETIS_MCG_FLL_FACTOR_2560
+#define KINETIS_MCG_FLL_FACTOR_FEE        KINETIS_MCG_FLL_FACTOR_2929
+#else
+#error "KINETIS_MCG_DCO_RANGE is wrong"
+#endif
+
+#ifndef KINETIS_MCG_USE_FAST_IRC
+#define KINETIS_MCG_USE_FAST_IRC          0
+#endif
+
+#if (KINETIS_MCG_USE_ERC == 0)
+#define KINETIS_MCG_USE_ERC               0
+#define KINETIS_MCG_USE_PLL               0
+#define KINETIS_MCG_ERC_FREQ              0
+#define KINETIS_MCG_ERC_FRDIV             0
+#define KINETIS_MCG_ERC_RANGE             0
+#define KINETIS_MCG_ERC_OSCILLATOR        0
+#endif
+
+#if (KINETIS_MCG_USE_PLL == 0)
+#define KINETIS_MCG_PLL_PRDIV             0
+#define KINETIS_MCG_PLL_VDIV0             0
+#define KINETIS_MCG_PLL_FREQ              0
+#endif
+
+#ifndef KINETIS_MCG_OSC_CLC
+#define KINETIS_MCG_OSC_CLC               0
+#endif
+
+#ifndef KINETIS_MCG_ERC_OSCILLATOR
+#error "KINETIS_MCG_ERC_OSCILLATOR not defined in periph_conf.h"
+#endif
+
+#ifndef KINETIS_MCG_ERC_FREQ
+#error "KINETIS_MCG_ERC_FREQ not defined in periph_conf.h"
+#endif
+
+#ifndef KINETIS_MCG_ERC_FRDIV
+#error "KINETIS_MCG_ERC_FRDIV not defined in periph_conf.h"
+#endif
+
+#ifndef KINETIS_MCG_ERC_RANGE
+#error "KINETIS_MCG_ERC_RANGE not defined in periph_conf.h"
+#endif
+
+#ifndef KINETIS_MCG_PLL_PRDIV
+#error "KINETIS_MCG_PLL_PRDIV not defined in periph_conf.h"
+#endif
+
+#ifndef KINETIS_MCG_PLL_VDIV0
+#error "KINETIS_MCG_PLL_VDIV0 not defined in periph_conf.h"
+#endif
+
+#ifndef KINETIS_MCG_PLL_FREQ
+#error "KINETIS_MCG_PLL_FREQ not defined in periph_conf.h"
+#endif
+
+static inline void kinetis_mcg_disable_pll(void)
+{
+    MCG->C5 = (uint8_t)0;
+    MCG->C6 = (uint8_t)0;
+}
+
+static inline void kinetis_mcg_set_ffl_factor(uint8_t factor)
```

Comment:
kinetis_mcg_set_ffl_factor -> kinetis_mcg_set_fll_factor, correct?

Could you add some short documentation for these internal functions?